crates/noirc_evaluator/src/ssa/integer.rs

```diff
@@ -229,7 +229,7 @@ fn block_overflow(
     // The instructions are insterted in a duplicate list( because of rust ownership..), which we use for
     // processing another cse round for the block because the truncates may be duplicated.
     let mut new_list = Vec::new();
-    let mut truncate_map = HashMap::new();
+    let mut truncate_map = BTreeMap::new();
```

Comment:
Could you add a comment explaining that we use a BTreeMap because the key order is necessary when we iterate over?

Reply:
Added